File: docs/src/testers/scraping/structured_data_response.py
# @sniptest filename=structured_data_response.py

from notte_sdk import NotteClient
from pydantic import BaseModel


class Product(BaseModel):
    name: str
    price: float


client = NotteClient()
url = "https://example.com/product"
product = client.scrape(url, response_format=Product)

# Access the extracted data
print(f"Name: {product.name}, Price: {product.price}")

File: docs/src/testers/scraping/structured_data_response.ts
import { z } from 'zod';
import { NotteClient } from 'notte-sdk';

const Product = z.object({
  name: z.string(),
  price: z.number(),
});

const client = new NotteClient({
  apiKey: process.env.NOTTE_API_KEY,
});

const url = 'https://example.com/product';
const product = await client.scrape(url, {
  response_format: Product,
});

// Access the extracted data
console.log(`Name: ${product.name}, Price: ${product.price}`);
